huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go

```diff
@@ -196,10 +229,37 @@ func resourceRdsReadReplicaInstanceCreate(ctx context.Context, d *schema.Resourc
 	instance := resp.Instance
 	d.SetId(instance.Id)
 	instanceID := d.Id()
-	if err := checkRDSInstanceJobFinish(client, resp.JobId, d.Timeout(schema.TimeoutCreate)); err != nil {
-		return diag.Errorf("error creating replica instance (%s): %s", instanceID, err)
+	// wait for order success
+	if resp.OrderId != "" {
+		bssClient, err := config.BssV2Client(config.GetRegion(d))
+		if err != nil {
+			return diag.Errorf("error creating BSS V2 client: %s", err)
+		}
+		if err := orders.WaitForOrderSuccess(bssClient, int(d.Timeout(schema.TimeoutCreate)/time.Second), resp.OrderId); err != nil {
+			return diag.Errorf("error waiting for replica order %s succuss: %s", resp.OrderId, err)
+		}
 	}
 
+	if resp.JobId != "" {
+		if err := checkRDSInstanceJobFinish(client, resp.JobId, d.Timeout(schema.TimeoutCreate)); err != nil {
+			return diag.Errorf("error creating replica instance (%s): %s", instanceID, err)
+		}
+	} else {
+		// for prePaid charge mode
+		stateConf := &resource.StateChangeConf{
+			Pending:      []string{"BUILD"},
+			Target:       []string{"ACTIVE", "BACKING UP"},
+			Refresh:      rdsInstanceStateRefreshFunc(client, instanceID),
+			Timeout:      d.Timeout(schema.TimeoutCreate),
+			Delay:        20 * time.Second,
+			PollInterval: 10 * time.Second,
+			// Ensure that the instance is 'ACTIVE', not going to enter 'BACKING UP'.
+			ContinuousTargetOccurence: 2,
+		}
+		if _, err = stateConf.WaitForState(); err != nil {
+			return diag.Errorf("error waiting for replica instance (%s) creation completed: %s", instanceID, err)
+		}
+	}
```

Comment:
pls using `common.WaitOrderResourceComplete()` method to instead, and bring into at end of following block:
```
if resp.OrderId != "" {
 ...
}
```

Reply:
done

Reply:
done